src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FC, MouseEvent, useState } from 'react'
+import styled, { css } from 'styled-components'
+import dayjs from 'dayjs'
+
+import { Theme, useTheme } from '../../hooks/useTheme'
+import { SecondaryButton } from '../Button'
+import { Icon } from '../Icon'
+import { CalendarTable } from './CalendarTable'
+import { YearPicker } from './YearPicker'
+
+type Props = {
+  from?: Date
+  to?: Date
+  onSelectDate: (e: MouseEvent, date: Date) => void
+  value?: Date
+}
+
+export const Calendar: FC<Props> = ({ from, to, onSelectDate, value }) => {
+  const themes = useTheme()
+  const now = dayjs()
+  const [currentMonth, setCurrentMonth] = useState(value ? dayjs(value) : now)
+  const [isSelectingYear, setIsSelectingYear] = useState(false)
+
+  const fromDay = dayjs(from || '1970-01-01')
+  const toDay = to ? dayjs(to) : now.add(50, 'year')
+  const prevMonth = currentMonth.subtract(1, 'month')
+  const nextMonth = currentMonth.add(1, 'month')
+
+  return (
+    <Container themes={themes}>
+      <Header themes={themes}>
+        <YearMonth>
+          {currentMonth.year()}年{currentMonth.month() + 1}月
+        </YearMonth>
+        <SecondaryButton onClick={() => setIsSelectingYear(!isSelectingYear)} size="s" square>
+          <Icon size={13} name={isSelectingYear ? 'fa-caret-up' : 'fa-caret-down'} />
+        </SecondaryButton>
+        <MonthButtonLayout>
+          <SecondaryButton
+            disabled={isSelectingYear || prevMonth.isBefore(fromDay, 'month')}
+            onClick={() => setCurrentMonth(prevMonth)}
+            size="s"
+            square
+          >
+            <Icon size={13} name="fa-chevron-left" />
+          </SecondaryButton>
+          <SecondaryButton
+            disabled={isSelectingYear || nextMonth.isAfter(toDay, 'month')}
+            onClick={() => setCurrentMonth(nextMonth)}
+            size="s"
+            square
+          >
+            <Icon size={13} name="fa-chevron-right" />
+          </SecondaryButton>
+        </MonthButtonLayout>
+      </Header>
+      <TableLayout>
+        <CalendarTable
+          current={currentMonth.toDate()}
+          from={fromDay.toDate()}
+          to={toDay.toDate()}
+          onSelectDate={onSelectDate}
+          selected={value}
+        />
+        {isSelectingYear && (
```

Comment:
`<CalendarTable>` and the `<YearPicker>` will be able to operated by keyboard. After that, there are issue yet, Keyboard user have to beyond `<CalendarTable>` that has many focusable element in order to use `<YearPicker>`.

If `<CalendarTable>` hide when displaying the `<YearPicker>` OR if you can put `<YearPicker>` before `<CalendarTable>`, this problem is solved.

In my opinion, It's better to swap the DOM order. Because it’s better to YearPicker's toggle button and itself are near by.